assets/src/edit-story/karma/fixture.js

```diff
@@ -0,0 +1,376 @@
+    // implemented via `waitFor`.
+    return Promise.resolve();
+  }
+
+  renderHook(func) {
+    return this._layoutStub.renderHook(func);
+  }
+
+  act(callback) {
+    return actPromise(callback);
+  }
+
+  querySelector(selector) {
+    return this._container.querySelector(selector);
+  }
+}
+
+/**
+ * A component stub. Allows two main features:
+ * 1. Mock a component's implementation.
+ * 2. Execute a hook against a component.
+ */
+class ComponentStub {
+  constructor(fixture, Component, matcher) {
+    this._fixture = fixture;
+    this._matcher = matcher;
+    this._implementation = null;
+
+    this._props = null;
+
+    let setRefresher;
+    this._refresh = () => {
+      act(() => {
+        if (setRefresher) {
+          setRefresher((v) => v + 1);
+        }
+      });
+    };
+
+    const pendingHooks = [];
+    this._pushPendingHook = (func) => {
+      let resolver;
+      const promise = new Promise((resolve) => {
+        resolver = resolve;
+      });
+      pendingHooks.push(() => {
+        const result = func();
+        resolver(result);
+      });
+      this._refresh();
+      return promise;
+    };
+
+    const Wrapper = forwardRef((props, ref) => {
+      this._props = props;
+
+      const [refresher, setRefresherInternal] = useState(0);
+      setRefresher = setRefresherInternal;
+      const hooks = useMemo(
+        () => {
+          const hooksToExecute = pendingHooks.slice(0);
+          pendingHooks.length = 0;
+          return hooksToExecute;
+        },
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+        [refresher]
+      );
+
+      const Impl = useMemo(
+        () => {
+          if (this._implementation) {
+            const MockImpl = forwardRef((fProps, fRef) =>
+              this._implementation(fProps, fRef)
+            );
+            MockImpl.displayName = `Stub(${
+              Component.displayName || Component.name || ''
+            })`;
+            return MockImpl;
+          }
+          return Component;
+        },
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+        [refresher]
+      );
+
+      return (
+        <HookExecutor key={refresher} hooks={hooks}>
+          <Impl _wrapped={true} ref={ref} {...props} />
+        </HookExecutor>
+      );
+    });
+    Wrapper.displayName = `Mock(${
+      Component.displayName || Component.name || ''
+    })`;
+    this._wrapper = Wrapper;
+  }
+
+  get and() {
+    return this;
+  }
+
+  get props() {
+    return this._props;
+  }
+
+  mockImplementation(implementation) {
+    this._implementation = implementation;
+    this._refresh();
+    return this;
+  }
+
+  callFake(implementation) {
+    return this.mockImplementation(implementation);
+  }
+
+  renderHook(func) {
+    return this._fixture.act(() => this._pushPendingHook(func));
+  }
+}
+
+function HookExecutor({ hooks, children }) {
+  hooks.forEach((func) => func());
+  return children;
+}
+
+class APIProviderFixture {
```

Comment:
This doesn't feel like true integration testing :/ Good for now though

Reply:
It's planned as an integration solution when editor separates from WP. The goal is to test how different parts of the editor coordinate, but, yeah, it's not a complete e2e.